### ISSUE TITLE
Research statistical packages for simulation

### ISSUE THREAD

**Issue by mmacfadden** [unbase/unbase#11]
We are going to want to be able to model user behavior, network bandwidth and latency, etc. We will probably want to use normal distributions for this. We should find a good pattern for sampling from a statistical distribution.

**Comment by SilverWingedSeraph** [comment]
I don't know that we need anything special for this; [`rand::distributions`](https://doc.rust-lang.org/rand/rand/distributions/index.html) has a wide selection, including [`Normal`](https://doc.rust-lang.org/rand/rand/distributions/normal/struct.Normal.html) and [Gaussian](https://doc.rust-lang.org/rand/rand/distributions/normal/struct.StandardNormal.html).

**Comment by mmacfadden** [comment]
Hi thanks. Yes I am aware of those packages. This is not just about the distribution. This is about how we build profiles for describing things like user behavior and network performance using these. Some design thought is needed to decide what parts of these concepts we need to model and to what fidelity.  Also which distributions are most appropriate for which concepts, etc.

**Comment by SilverWingedSeraph** [comment]
Oh, I see. That makes more sense. I know there are packages around for data
mocking but it might make sense to roll out own depending on exactly what
we need.

**Comment by mmacfadden** [comment]
Yeah at this point I am not sure what we want to "use". Totally focused on what we need to simulate and then thinking what kinds of statistics we need.

**Comment by dnorman** [comment]
@mmacfadden @SilverWingedSeraph I'm hoping it'll become a little clearer what we need to do once I've completed the upgrades to the network simulator which @mmacfadden and I had discussed the other night. I have substantially more clarity about what we'll need to support the user behavior simulator. Should be another couple days on that. Working on other core stuff tonight :)
